fossunited/api/profile.py

```diff
@@ -78,17 +77,23 @@ def update_profile(fields_dict):
             "mastodon": fields_dict.get("mastodon"),
         }
 
-        frappe.db.set_value(USER_PROFILE, user_doc.name, updated_fields)
+        profile = frappe.get_doc(USER_PROFILE, user_doc.name)
+        for field, value in updated_fields.items():
+            if hasattr(profile, field):
+                setattr(profile, field, value)
+        profile.save(ignore_permissions=True)
```

Comment:
not sure why this pattern is better than the previous pattern

Reply:
Do you mean the `ignore permissions` & `save` method? They enable us to run [document controller methods](https://frappeframework.com/docs/user/en/basics/doctypes/controllers) .

Basically, with previous database api, i would not be able to run any methods that are written [here](https://github.com/fossunited/fossunited/blob/develop/fossunited/foss_profiles/doctype/foss_user_profile/foss_user_profile.py) but with document api, we can run those methods as they were originally intended : whether on save, or on update of any info